mesos/framework/mirror_maker_task.go

```diff
@@ -0,0 +1,118 @@
+package framework
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	mesos "github.com/mesos/mesos-go/mesosproto"
+	util "github.com/mesos/mesos-go/mesosutil"
+)
+
+type MirrorMakerTask struct {
+	*CommonTask
+}
+
+func NewMirrorMakerTask(id string, queryParams url.Values) (*MirrorMakerTask, error) {
+	taskData := &TaskData{
+		ID:    id,
+		State: TaskStateInactive,
+		Config: TaskConfig{
+			"num.producers": "1",
+			"num.streams":   "1",
+			"queue.size":    "10000",
+		},
+	}
+
+	err := taskData.Update(queryParams)
+	if err != nil {
+		return nil, err
+	}
+
+	return &MirrorMakerTask{
+		&CommonTask{TaskData: taskData},
+	}, nil
+}
+
+func (mm *MirrorMakerTask) NewTaskInfo(offer *mesos.Offer) *mesos.TaskInfo {
+	taskName := fmt.Sprintf("mirrormaker-%s", mm.ID)
+	taskId := &mesos.TaskID{
+		Value: proto.String(fmt.Sprintf("%s-%s", taskName, uuid())),
+	}
+
+	data, err := json.Marshal(mm.Config)
+	if err != nil {
+		panic(err)
+	}
+
+	taskInfo := &mesos.TaskInfo{
+		Name:     proto.String(taskName),
+		TaskId:   taskId,
+		SlaveId:  offer.GetSlaveId(),
+		Executor: mm.createExecutor(),
+		Resources: []*mesos.Resource{
+			util.NewScalarResource("cpus", mm.Cpu),
+			util.NewScalarResource("mem", mm.Mem),
+		},
+		Data: data,
+	}
+
+	return taskInfo
+}
+
+func (mm *MirrorMakerTask) String() string {
+	response := "    type: mirrormaker\n"
```

Comment:
this should be another interface so that folks can do their own thing or use what we have packaged for them ya?